c10/hammerblade/emul/kernel_trampoline.cpp

```diff
@@ -1,11 +1,11 @@
 #include <kernel_trampoline.h>
 
-std::map<std::string, std::function<int(uint32_t, uint32_t*)>> kernelMap;
-std::vector<std::function<int(uint32_t, uint32_t*)>> enqueued_kernel;
+std::map<std::string, std::function<int(uint32_t, uint64_t*)>> kernelMap;
+std::vector<std::function<int(uint32_t, uint64_t*)>> enqueued_kernel;
 std::vector<uint32_t>  enqueued_argc;
-std::vector<uint32_t*> enqueued_argv;
+std::vector<uint64_t*> enqueued_argv;
```

Comment:
Doesn't this increase the chance of mismatch between cosim and emulation? If it makes it easy to maintain emulation code, it makes sense though... just curious.

Reply:
Yes your concern is more than valid. The issue I was facing is that, it is really hard to request a large chunk of low address memory (>2M), even with `mmap`. If we want to run MNIST then we certainly needs a buffer larger than that. So I changed the definition of `bsg_tensor_t` as well, with a macro to choose between `uint32_t` and `uint64_t`.